lib/msal-common/src/authority/Authority.ts

```diff
@@ -438,33 +443,46 @@ export class Authority {
      * @param newMetadata
      */
     private async updateCloudDiscoveryMetadata(metadataEntity: AuthorityMetadataEntity): Promise<AuthorityMetadataSource> {
+        this.logger.verbose("Attempting to get cloud discovery metadata in the config");
         let metadata = this.getCloudDiscoveryMetadataFromConfig();
         if (metadata) {
+            this.logger.verbose(`Found the following cloud discovery metata in the config: ${JSON.stringify(metadata)}`);
             metadataEntity.updateCloudDiscoveryMetadata(metadata, false);
             return AuthorityMetadataSource.CONFIG;
         }
+        this.logger.verbose("Did not find cloud discovery metadata in the config");
 
-        // If The cached metadata came from config but that config was not passed to this instance, we must go to the network
+        // If the cached metadata came from config but that config was not passed to this instance, we must go to the network
+        this.logger.verbose("Attempting to get cloud discovery metadata in the cache");
         if (this.isAuthoritySameType(metadataEntity) && metadataEntity.aliasesFromNetwork && !metadataEntity.isExpired()) {
+            this.logger.verbose(`Found the following metadata in the cache: ${JSON.stringify(metadata)}`);
```

Comment:
Since this is on the hot path, we should probably not create `${JSON.stringify(metadata)}` every time. Is there a way to do something like: 

```
if (this.logger.LogLevel == LogLevel.Verbose)
{
    this.logger.verbose(`Found the following metadata in the cache: ${JSON.stringify(metadata)}`);
}
```

Reply:
I know we added a trace logger for this purpose in browser. I think we can do an equivalent here. In Node it'll be a bit more cumbersome since you can't collapse printed logs in the console like you can with browser dev tools.

Reply:
Is this method, `updateCloudDiscoveryMetadata`, really in a hot path? Does it get called for every request? (I'd assume instance discovery would be done once per app.)
If it is in the hot path, we could log only something like canonical authority from the metadata.